src/gui/MouseSelector.cc

```diff
@@ -85,8 +89,11 @@ int MouseSelector::select(const Renderer *renderer, int x, int y) {
 
   // TODO: Ideally, we should make the above configurable and reduce duplicate render code in this function.
 
-  if (x > static_cast<int>(this->view->cam.pixel_width) || x < 0 ||
-      y > static_cast<int>(this->view->cam.pixel_height) || y < 0) {
+  const auto width = this->view->cam.pixel_width;
+  const auto height = this->view->cam.pixel_height;
+
+  if (x >= static_cast<int>(width) || x < 0 ||
+      y >= static_cast<int>(height) || y < 0) {
```

Comment:
I'm not sure to understand the use of auto + static_cast
```suggestion
  const int width = this->view->cam.pixel_width;
  const int height = this->view->cam.pixel_height;

  if (x >= width || x < 0 || y >= height || y < 0) {
```

Reply:
Yeah, make sense. Thx.